packages/ember-data/lib/system/relationship-meta.js

```diff
@@ -16,13 +16,24 @@ export function typeForRelationshipMeta(store, meta) {
   return type;
 }
 
+export function generateLink(store, kind, type) {
```

Comment:
Not sure this makes a difference? You will call the function every time you want to generate a link no?